src/py/mat3ra/made/tools/analyze/slab.py

```diff
@@ -0,0 +1,108 @@
+        slab_build_configuration_dict = metadata.build.configuration
+        if slab_build_configuration_dict.get("type") != "SlabConfiguration":
+            raise ValueError("Material is not a slab.")
+
+        return SlabConfiguration(**slab_build_configuration_dict)
+
+    def get_slab_with_additional_layers_configurations(
+        self,
+        additional_layers: Union[int, float] = 1,
+        vacuum_thickness: float = 5.0,
+    ) -> Tuple[SlabWithAdditionalLayersConfiguration, SlabConfiguration]:
+        """
+        Analyze a slab material and create two configurations:
+        - with original number of layers and adjusted vacuum to match the new height
+        - with additional layers and the vacuum thickness on top.
+        Returns:
+            (SlabWithAdditionalLayersConfiguration, SlabConfiguration)
+        """
+        crystal, miller_indices, original_layers, termination_formula = self._get_slab_parameters()
+
+        # Calculate the vacuum thickness needed for the original slab to match the height of the slab with additional layers
+        vacuum_to_match_height = self._vacuum_thickness_for_original_slab(
+            additional_layers=additional_layers,
+            vacuum_thickness=vacuum_thickness,
+        )
+
+        configuration_with_added_layers = SlabWithAdditionalLayersConfiguration.from_parameters(
+            material_or_dict=crystal,
+            miller_indices=miller_indices,
+            number_of_layers=original_layers,
+            number_of_additional_layers=additional_layers,
+            termination_formula=termination_formula,
+            vacuum=vacuum_thickness,
+        )
+        configuration_original_with_adjusted_vacuum = SlabConfiguration.from_parameters(
+            material_or_dict=crystal,
+            miller_indices=miller_indices,
+            number_of_layers=original_layers,
+            termination_formula=termination_formula,
+            vacuum=vacuum_to_match_height,
+        )
+
+        return (configuration_with_added_layers, configuration_original_with_adjusted_vacuum)
+
+    def _get_slab_parameters(self):
+        """
+        Extract common slab parameters from the current slab configuration.
+        Returns:
+            Tuple of (crystal, miller_indices, original_layers, termination_formula)
+        """
+        slab_configuration = self.get_slab_configuration()
+        atomic_layers = slab_configuration.atomic_layers
+        original_layers = atomic_layers.number_of_repetitions
+        crystal = atomic_layers.crystal
+        miller_indices = atomic_layers.miller_indices
+        termination_top = atomic_layers.termination_top
+
+        if isinstance(termination_top, dict):
+            termination_formula = termination_top.get("chemical_elements")
+        else:
+            termination_formula = termination_top.formula
+
+        return crystal, miller_indices, original_layers, termination_formula
```

Comment:
This should be coming from `atomic_layers.get_slab_parameters()` or `to_parameters`